upup/pkg/fi/cloudup/vsphere/vsphere_cloud.go

```diff
@@ -66,6 +97,78 @@ func (c *VSphereCloud) DNS() (dnsprovider.Interface, error) {
 }
 
 func (c *VSphereCloud) FindVPCInfo(id string) (*fi.VPCInfo, error) {
-	glog.Warningf("FindVPCInfo not (yet) implemented on VSphere")
+	glog.Warning("FindVPCInfo not (yet) implemented on VSphere")
 	return nil, nil
 }
+
+func (c *VSphereCloud) CreateLinkClonedVm(vmName, vmImage *string) (string, error) {
+	f := find.NewFinder(c.Client.Client, true)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dc, err := f.Datacenter(ctx, c.Datacenter)
+	if err != nil {
+		return "", err
+	}
+	f.SetDatacenter(dc)
+
+	templateVm, err := f.VirtualMachine(ctx, *vmImage)
+	if err != nil {
+		return "", err
+	}
+
+	datacenterFolders, err := dc.Folders(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// Create snapshot of the template VM if not already snapshotted.
+	snapshot, err := createSnapshot(ctx, templateVm)
+	if err != nil {
+		return "", err
+	}
+
+	clsComputeRes, err := f.ClusterComputeResource(ctx, c.Cluster)
+	glog.V(4).Infof("clsComputeRes is %s", clsComputeRes)
```

Comment:
Nit: Logging variable name might not be very useful. Replace with ClusterComputeResource?
Same for other places.